packages/local_auth/local_auth_windows/windows/local_auth_plugin.cpp

```diff
@@ -0,0 +1,223 @@
+  if (method_call.method_name().compare("authenticate") == 0) {
+    Authenticate(method_call, std::move(result));
+  } else if (method_call.method_name().compare("getAvailableBiometrics") == 0) {
+    GetAvailableBiometrics(std::move(result));
+  } else if (method_call.method_name().compare("isDeviceSupported") == 0) {
+    IsDeviceSupported(std::move(result));
+  } else {
+    result->NotImplemented();
+  }
+}
+
+// Starts authentication process
+winrt::fire_and_forget LocalAuthPlugin::Authenticate(
+    const MethodCall<EncodableValue>& method_call,
+    std::unique_ptr<MethodResult<EncodableValue>> result) {
+  auto reasonW = Utf16FromUtf8(GetArgument<std::string>(
+      "localizedReason", method_call.arguments(), std::string()));
+
+  auto biometricOnly =
+      GetArgument<bool>("biometricOnly", method_call.arguments(), false);
+  if (biometricOnly) {
+    result->Error("biometricOnlyNotSupported",
+                  "Windows doesn't support the biometricOnly parameter.");
+    co_return;
+  }
+
+  auto ucvAvailability =
+      co_await user_consent_verifier_->CheckAvailabilityAsync();
+
+  if (ucvAvailability ==
+      Windows::Security::Credentials::UI::UserConsentVerifierAvailability::
+          DeviceNotPresent) {
+    result->Error("NoHardware", "No biometric hardware found");
+    co_return;
+  } else if (ucvAvailability ==
+             Windows::Security::Credentials::UI::
+                 UserConsentVerifierAvailability::NotConfiguredForUser) {
+    result->Error("NotEnrolled", "No biometrics enrolled on this device.");
+    co_return;
+  } else if (ucvAvailability !=
+             Windows::Security::Credentials::UI::
+                 UserConsentVerifierAvailability::Available) {
+    result->Error("NotAvailable", "Required security features not enabled");
+    co_return;
+  }
+
+  try {
+    auto consentResult =
+        co_await user_consent_verifier_->RequestVerificationForWindowAsync(
+            reasonW);
+
+    result->Success(EncodableValue(
+        consentResult == Windows::Security::Credentials::UI::
+                             UserConsentVerificationResult::Verified));
+  } catch (...) {
+    result->Success(EncodableValue(false));
+  }
+}
+
+// Returns biometric types available on device
+winrt::fire_and_forget LocalAuthPlugin::GetAvailableBiometrics(
+    std::unique_ptr<MethodResult<EncodableValue>> result) {
+  try {
+    flutter::EncodableList biometrics;
+    auto ucvAvailability =
+        co_await user_consent_verifier_->CheckAvailabilityAsync();
+    if (ucvAvailability == Windows::Security::Credentials::UI::
+                               UserConsentVerifierAvailability::Available) {
+      biometrics.push_back(EncodableValue("fingerprint"));
+      biometrics.push_back(EncodableValue("face"));
+      biometrics.push_back(EncodableValue("iris"));
+    }
+    result->Success(biometrics);
+  } catch (const std::exception& e) {
+    result->Error("no_biometrics_available", e.what());
```

Comment:
Does this case not map to one of the established error codes?

Reply:
I'm using the same as the Android one:
https://github.com/flutter/plugins/blob/bde713a230d3931f17105f2b4f8439cf448d031a/packages/local_auth/local_auth_android/android/src/main/java/io/flutter/plugins/localauth/LocalAuthPlugin.java#L269

Reply:
Ah. That's probably a bug, but since we clearly need to do a sweep to enforce more consistency already, matching Android is fine.